service/frontend/nexus_http_handler.go

```diff
@@ -320,10 +321,14 @@ func (h *NexusHTTPHandler) parseTlsAndAuthInfo(r *http.Request, nc *nexusContext
 		r = r.WithContext(h.auth.EnhanceContext(r.Context(), authInfo, nc.claims))
 	}
 
-	return r.WithContext(context.WithValue(r.Context(), nexusContextKey{}, *nc)), nil
+	return r, nil
 }
 
-func (h *NexusHTTPHandler) serveResolvedURL(w http.ResponseWriter, r *http.Request, u *url.URL) {
+func (h *NexusHTTPHandler) serveResolvedURL(w http.ResponseWriter, r *http.Request, u *url.URL, nc *nexusContext) {
+	// Attach Nexus context to response writer and request context.
+	w = newNexusHTTPResponseWriter(w, nc)
+	r = r.WithContext(context.WithValue(r.Context(), nexusContextKey{}, *nc))
```

Comment:
Let's just put the point on the context instead (and make sure to extract it as a pointer).